### PR TITLE
Ensure that the /bin/terraform-provider-matchbox binary is present

### PR DESCRIPTION
**Pull request by kirankt**
The terraform-provider-matchbox binary was not being correctly extracted into the /bin directory. This PR ensures that the binary is moved to /bin correctly.

This PR also ensures the changes made in this PR and #5762 are reconciled between the two UPI images.

Fixes issues see here: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/27383/rehearse-27383-pull-ci-openshift-installer-master-e2e-metal-upi/1516093307896205312/build-log.txt

### CODE REVIEW

File: images/installer/Dockerfile.upi.ci.rhel8 [hunk 1]
```diff
@@ -42,13 +42,13 @@ RUN yum update -y && \
 RUN pip-2 install pyopenssl
 ENV CLOUDSDK_PYTHON=/usr/bin/python
 
-ENV TERRAFORM_VERSION=0.12.24
+ENV TERRAFORM_VERSION=1.0.11
```
Comment (patrickdillon) [inline images/installer/Dockerfile.upi.ci.rhel8:45]: Can you move this into a separate commit and write a commit message for each.

Comment (kirankt) [reply images/installer/Dockerfile.upi.ci.rhel8:45]: Done. The change to the RHEL Dockerfile was just to gain parity with the main Dockerfile.

Comment (patrickdillon) [reply images/installer/Dockerfile.upi.ci.rhel8:45]: I meant the terraform version, that is unrelated to the matchbox version, right? 

Can you split the terraform version into a different commit, unless it is related. If it is related please note that in the commit message. 

Can you write a commit message for each like https://github.com/openshift/installer/blob/master/CONTRIBUTING.md#commit-message-format

Comment (kirankt) [reply images/installer/Dockerfile.upi.ci.rhel8:45]: This was originally changed here: https://github.com/openshift/installer/pull/5762/files and the reconciliation didn't happen on the RHEL Dockerfile in that PR. I added it here.

In any case, I'll redo the commit as you pointed out.

Comment (patrickdillon) [reply images/installer/Dockerfile.upi.ci.rhel8:45]: Got it. We should have caught it in that PR review. Thanks for taking care of it here!
